static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -0,0 +1,164 @@
+import {mat3} from 'gl-matrix';
+
+import {Flamegraph} from '../flamegraph';
+import {FlamegraphTheme} from '../flamegraph/FlamegraphTheme';
+import {FlamegraphFrame} from '../flamegraphFrame';
+import {
+  ELLIPSIS,
+  findRangeBinarySearch,
+  getContext,
+  Rect,
+  resizeCanvasToDisplaySize,
+  trimTextCenter,
+} from '../gl/utils';
+
+export function isOutsideView(frame: Rect, view: Rect, inverted: boolean): boolean {
+  // Frame is outside of the view on the left
+  if (frame.right <= view.left) {
+    return true;
+  }
+
+  // Frame is outside of the view on the right
+  if (frame.left >= view.right) {
+    return true;
+  }
+
+  // Frame is above the view
+  if (frame.bottom <= view.top) {
+    return true;
+  }
+
+  // Frame is below the view
+  if (frame.top >= view.bottom) {
+    return true;
+  }
+
+  // @TODO check if we still need this
+  if (inverted) {
+    if (frame.top - 1 >= view.bottom) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+class TextRenderer {
+  textCache: Record<string, number> = {};
+
+  canvas: HTMLCanvasElement;
+  context: CanvasRenderingContext2D;
+  theme: FlamegraphTheme;
+  flamegraph: Flamegraph;
+
+  constructor(canvas: HTMLCanvasElement, flamegraph: Flamegraph, theme: FlamegraphTheme) {
+    this.canvas = canvas;
+    this.theme = theme;
+    this.flamegraph = flamegraph;
+
+    this.context = getContext(canvas, '2d');
+
+    resizeCanvasToDisplaySize(canvas);
+  }
+
+  clearCache(): void {
+    this.textCache = {};
+  }
+
+  measureText(context: CanvasRenderingContext2D, text: string): number {
+    if (this.textCache[text]) {
+      return this.textCache[text];
+    }
+    this.textCache[text] = context.measureText(text).width;
+    return this.textCache[text];
+  }
+
+  draw(configViewSpace: Rect, configSpace: Rect, configToPhysicalSpace: mat3): void {
+    this.context.font = `${
+      this.theme.SIZES.BAR_FONT_SIZE * window.devicePixelRatio
+    }px "Source Code Pro", Courier, monospace`;
```

Comment:
Should this be defined inside of the `FlamegraphTheme`? Perhaps the list of fonts at least? That way, it'll be easier to change all of them to conform to Sentry's styles.